Add DataFrame.persist, and notes on execution model (#307)
data-apis/dataframe-api: spec/purpose_and_scope.md

Implementation details of the dataframes and execution of operations. This includes:

- How data is represented and stored (whether the data is in memory, disk, distributed)

> - Expectations on when the execution is happening (in an eager or lazy way), other than `DataFrame.persist`

Reviewer: minor: not entirely accurate, since it's only a hint so there is still no "when" prescribed.

How about saying instead: _"(see `Exection model` for some caveats)"_ in order to keep things in one place?

> - Expectations on when the execution is happening (in an eager or lazy way), other than `DataFrame.persist`

Author: ```suggestion
- Expectations on when the execution is happening (in an eager or lazy way) (see `execution model` for some caveats)
```
